poollet/volumepoollet/controllers/volumepool_controller.go

```diff
@@ -113,10 +117,11 @@ func (r *VolumePoolReconciler) reconcile(ctx context.Context, log logr.Logger, v
 	}
 
 	log.V(1).Info("Reconciled")
-	return ctrl.Result{}, nil
+	return ctrl.Result{RequeueAfter: r.RelistPeriod}, nil
```

Comment:
Simply requeueing after a relist period will solve the immediate symptom, however it comes at the cost of any part of the reconciliation of the `VolumePool` that is not subject to volume classes being executed additionally at each reconciliation. A solution that really fixes the problem is to give the `VolumeClassMapper` the possibility to emit events that can then be used as a `source.Source` in the `SetupWithManager` method.